### PR TITLE
[IDE] Fix visiting DeferStmt

### PR DESCRIPTION
**Pull request by rintaro**
* Call balanced `walkToStmtPost()` for DeferStmt.
* In `walkToStmtPre()`, return `DeferStmt` itself instead of the body.

rdar://problem/39948168

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/IDE/range_info_basics.swift [hunk 1]
```diff
@@ -144,6 +144,11 @@ func testExpressionVariables() -> Int {
   return bar()
 }
 
+func testDefer() {
+  defer {
+  }
+}
```
Comment (nathawes) [inline test/IDE/range_info_basics.swift]: @rintaro does this test fail without your changes? I thought I'd tried this one when I was reducing the original test case with no luck, but might be misremembering.

**Comment by rintaro** [comment]
Probably the minimum reproducer is:
```
$ cat test.swift
defer {}

$ ../build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/swift-ide-test -range -pos=1:1 -end-pos=1:2 -source-filename test.swift
Assertion failed: (ContextStack.back().Parent.getOpaqueValue() == Node.getOpaqueValue()), function leave, file /Users/rintaro/Repositories/swift-oss/swift/lib/IDE/SwiftSourceDocInfo.cpp, line 1120.
0  swift-ide-test           0x0000000107a3cb68 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 40
1  swift-ide-test           0x0000000107a3d256 SignalHandler(int) + 486
2  libsystem_platform.dylib 0x00007fff6b54df5a _sigtramp + 26
3  swift-ide-test           0x0000000108c3f4c0 cmark_strbuf__initbuf + 104888
4  libsystem_c.dylib        0x00007fff6b2eb1ae abort + 127
5  libsystem_c.dylib        0x00007fff6b2b31ac basename_r + 0
6  swift-ide-test           0x0000000104575d37 swift::ide::RangeResolver::Implementation::leave(swift::ASTNode) + 1015
7  swift-ide-test           0x0000000104575d82 swift::ide::RangeResolver::walkToStmtPost(swift::Stmt*) + 34
8  swift-ide-test           0x000000010459b287 (anonymous namespace)::SemaAnnotator::walkToStmtPost(swift::Stmt*) + 23
9  swift-ide-test           0x0000000104ac7b85 (anonymous namespace)::Traversal::doIt(swift::Decl*) + 501
10 swift-ide-test           0x0000000104ac797b swift::Decl::walk(swift::ASTWalker&) + 27
11 swift-ide-test           0x0000000104b73c6a swift::SourceFile::walk(swift::ASTWalker&) + 170
12 swift-ide-test           0x000000010459a297 swift::SourceEntityWalker::walk(swift::SourceFile&) + 119
13 swift-ide-test           0x0000000104575fb2 swift::ide::RangeResolver::resolve() + 50
14 swift-ide-test           0x0000000104494894 main + 28868
15 libdyld.dylib            0x00007fff6b23f015 start + 1
16 libdyld.dylib            0x0000000000000006 start + 2497449970
Stack dump:
0.	Program arguments: ../build/Ninja-ReleaseAssert/swift-macosx-x86_64/bin/swift-ide-test -range -pos=1:1 -end-pos=1:2 -source-filename test.swift 
1.	While walking into decl declaration 0x7ff6d4064638 at test.swift:1:1
Abort trap: 6
```

**Comment by nathawes** [comment]
Oh sorry, I think I was including the function's closing `}` when I tried this, which seems to be the difference. Thanks for checking!

**Review by nathawes — APPROVED** [review]
LGTM!

**Pull request merged**